### ISSUE TITLE
Python 3.11 needs `find_spec` implemented in JuliaImporter

### ISSUE THREAD

**Issue by MilesCranmer** [JuliaPy/pyjulia#509]
With Python 3.11, `JuliaImporter` should implement `find_spec`, and `JuliaModuleLoader` should implement `exec_module`. Otherwise, the following warnings appear:

```
<frozen importlib._bootstrap>:1049: ImportWarning: JuliaImporter.find_spec() not found; falling back to find_module()
<frozen importlib._bootstrap>:673: ImportWarning: JuliaModuleLoader.exec_module() not found; falling back to load_module()
```

Shouldn't be too bad to implement though.

**Comment by MilesCranmer** [comment]
Yep, I think this is breaking PyJulia on 3.12 @mkitti. Seems like we need to implement `find_spec` (https://docs.python.org/3/library/importlib.html#importlib.abc.MetaPathFinder.find_spec) in `JuliaImporter` as well as `exec_module` in `JuliaModuleLoader`.

**Comment by MilesCranmer** [comment]
Fixed from #538
